src/java/org/apache/cassandra/net/OutboundSink.java

```diff
@@ -56,6 +57,24 @@ public void accept(Message<?> message, InetAddressAndPort to, ConnectionType con
         }
     }
 
+    private static class PostSink implements Sink
+    {
+        final BiConsumer<Message<?>, InetAddressAndPort> postSink;
+        final Sink sink;
+
+        private PostSink(BiConsumer<Message<?>, InetAddressAndPort> postSink, Sink sink)
+        {
+            this.postSink = postSink;
+            this.sink = sink;
+        }
+
+        public void accept(Message<?> message, InetAddressAndPort to, ConnectionType connectionType)
+        {
+            sink.accept(message, to, connectionType);
+            postSink.accept(message, to);
```

Comment:
Yes, this is intentional because it models how the corresponding code works in DSE in the original port. But also, I don't think  there's a way to detect is `sink.accept` fails -- it doesn't return anything.

Reply:
i was thinking about an exception